src/main/java/com/sap/sgs/phosphor/fosstars/model/score/AbstractScore.java

```diff
@@ -240,4 +252,50 @@ static ScoreValue calculateIfNecessary(Score score, ValueHashSet values) {
     throw new IllegalArgumentException(String.format(
         "Hey! I expected a ScoreValue for a score but got %s!", value.getClass()));
   }
+
+  /**
+   * Checks if all values are unknown.
+   *
+   * @param values The values to be checked.
+   * @return True if all values are unknown, false otherwise.
+   * @throws IllegalArgumentException If values are empty.
+   */
+  protected static boolean allUnknown(Value... values) {
```

Comment:
I would name this method as areAllUnknown()

Reply:
I would do that as well if it was not a static method. Then, client code would look like this:

```java
if (values.areAllUnknown()) {
```

Here we have a static method, and client code looks like the following:

```java
if (allUnknown(values)) {
```

It looks fine to me. Please let me know if you see any issue here. Otherwise, let's please keep it as-is.

Reply:
OK )